### PR TITLE
ENH: keep the attributes around

### PR DESCRIPTION
**Pull request by martinfleis**
Ton of tests failing but the attributes are preserved.

### CODE REVIEW

**Comment by martinfleis** [comment]
@jGaboardi can you have a look at the failures and suggest optimal solutions form your perspective?

**Comment by martinfleis** [comment]
Well, it should not change the geometry... I'll have a look. I more thought about the missing column in appalochicola expected.

**Comment by jGaboardi** [comment]
> I more thought about the missing column in appalochicola expected.

Not sure. I'll take a look in a bit.

**Comment by jGaboardi** [comment]
> I more thought about the missing column in appalochicola expected.

* That column is missing because that "truth" hasn't been updated.
  * [`apalachicola_simplified_standard.parquet`](https://github.com/uscuni/sgeop/blob/main/sgeop/tests/data/apalachicola_simplified_standard.parquet)
  * [`apalachicola_simplified_exclusion_mask.parquet`](https://github.com/uscuni/sgeop/blob/main/sgeop/tests/data/apalachicola_simplified_exclusion_mask.parquet) 
* Although the Apalachicola test case doesn't hit nearly as many lines of the code base, it is nice because it passes and can be written from any OS (unlike our 6 large FUA tests).
 * I will see about getting those updated to find out where that gets us.
 * Since the appalachicola tests are failing [here](https://github.com/uscuni/sgeop/blob/ed6bf40c2e2766b75b0dd7a99367234a6bb8ae0f/sgeop/tests/test_simplify.py#L67) (after the [length check](https://github.com/uscuni/sgeop/blob/ed6bf40c2e2766b75b0dd7a99367234a6bb8ae0f/sgeop/tests/test_simplify.py#L64)) it appears they are not affected by the altered functionality that leads to differences in geometry.

**Comment by jGaboardi** [comment]
* Keep in mind that the ["full" CI suite is only run on non-oldest Ubuntu](https://github.com/uscuni/sgeop/blob/ed6bf40c2e2766b75b0dd7a99367234a6bb8ae0f/sgeop/tests/test_simplify.py#L99-L101).
* When we check that – for example [`ubuntu-latest, py313_sgeop-latest`](https://github.com/uscuni/sgeop/actions/runs/12138536851/job/33844197819) – we see that all tests are failing within [`test_simplify.py`](https://github.com/uscuni/sgeop/blob/main/sgeop/tests/test_simplify.py)
  * For small Appalachicola, it's the columns – passes length check; seems like no geometries are affected
  * For  the full FUA's its a mixed bag:
    * [`aleppo_1133`](https://github.com/uscuni/sgeop/actions/runs/12138536851/job/33844197819#step:6:814) – passes [length check](https://github.com/uscuni/sgeop/blob/ed6bf40c2e2766b75b0dd7a99367234a6bb8ae0f/sgeop/tests/test_simplify.py#L96); fails [`'_status'` check](https://github.com/uscuni/sgeop/blob/ed6bf40c2e2766b75b0dd7a99367234a6bb8ae0f/sgeop/tests/test_simplify.py#L100)
    * [`auckland_869`](https://github.com/uscuni/sgeop/actions/runs/12138536851/job/33844197819#step:6:792) – fails length check
    * [`bucaramanga_4617`](https://github.com/uscuni/sgeop/actions/runs/12138536851/job/33844197819#step:6:821) – fails length check
    * [`douala_809`](https://github.com/uscuni/sgeop/actions/runs/12138536851/job/33844197819#step:6:804) – passes length check; fails index check – seems to be 1 *more* edge in the observed
    * [`liege_1656`](https://github.com/uscuni/sgeop/actions/runs/12138536851/job/33844197819#step:6:797) – passes length check; fails `'_status'` check
    * [`slc_4881`](https://github.com/uscuni/sgeop/actions/runs/12138536851/job/33844197819#step:6:809) – passes length check; fails index check – seems to be 3 *fewer* edges in the observed

**Comment by jGaboardi** [comment]
I'll also add some language in [the docstring](https://github.com/uscuni/sgeop/blob/ed6bf40c2e2766b75b0dd7a99367234a6bb8ae0f/sgeop/simplify.py#L653-L663) to reflect this enhancement (once we figure out where geometry altering is happening).

Something along the lines of:
> "... the attributes of the input network data are preserved where possible through the simplification process in modified geometries... ... entirely new geometries that are created thorough the simplification process (e.g., Voronoi skeletons) will have no attributes..."

**Comment by martinfleis** [comment]
I still don't understand how has this affected number of edges... something for tomorrow.

**Comment by jGaboardi** [comment]
Doing some digging now – within the Apalachicola Standard (non-exclusion mask) stuff there are 74 mismatches in the `'_status'` column – all `'changed'` in known and `'original'` in observed.

**Comment by jGaboardi** [comment]
* known = thick & transparent-ish
* observed = thin & bold
gh_132_known_vs_observed_apalachicola

File: sgeop/simplify.py [hunk 1]
```diff
@@ -265,12 +266,20 @@ def simplify_singletons(
         new["_status"] = "new"
         new.geometry = new.simplify(max_segment_length * simplification_factor)
         new_roads = pd.concat([cleaned_roads, new], ignore_index=True)
+        agg = {"_status": _status}
+        for c in cleaned_roads.columns.drop(cleaned_roads.active_geometry_name):
+            if c != "_status":
```
Comment (jGaboardi) [inline sgeop/simplify.py:271]: @martinfleis this issue was is these chunks where we are adding the `agg` dict, the initial `{"_status": _status}` was being overwritten with `{"_status": 'first'}`. That was wreaking havoc on the results.

**Comment by jGaboardi** [comment]
* I found the culprit of the weirdness, but something is still happening whereby all non-[geometry, _status] columns are being dropped (in the FUA tests only?) somewhere in the [second loop](https://github.com/uscuni/sgeop/blob/c30a9aee32968ed8612485a40ac551a6696946b2/sgeop/simplify.py#L790-L800).
* ***This means the current green CI is misleading*** -- attributes are not being retained and "passing" is actually failing 
* The FUA tests should all fail, then we updated the expected FUA results with fresh truth.
* This problem is not affecting the Apalachicola scenarios

--------------------------

Also, we should check and update the "original" FUA parquet files for uniformity. Should have index of `['u', 'v', 'key']` while others have `['u', 'v', '__key']`.

File: sgeop/tests/test_simplify.py [hunk 2]
```diff
@@ -87,6 +87,7 @@ def test_simplify_network_full_fua(aoi, tol, known_length):
         geopandas.read_parquet(full_fua_data / aoi / "original.parquet")
     )
     observed_length = observed.geometry.length.sum()
+    assert "highway" in observed.columns
```
Comment (martinfleis) [inline sgeop/tests/test_simplify.py:90]: @jGaboardi I am wondering if we want to update the known parquets to contain all the other columns and test their equality or if this is enough.

**Review by martinfleis — COMMENTED** [review]
All should be fixed now. Manually checked on all our case studies.

**Comment by martinfleis** [comment]
I don't understand the typing issue :D

File: sgeop/tests/test_simplify.py (hunk 2 above)
Comment (jGaboardi) [reply sgeop/tests/test_simplify.py:90]: I think this is enough for now, maybe with #135 we also can reset the indices so the parquet also have `['u', 'v', 'key']`. That's probably enough, right?

**Comment by jGaboardi** [comment]
> I don't understand the typing issue :D

Typing fixed.

**Review by jGaboardi — APPROVED** [review]

**Pull request merged**